internal/internal.go

```diff
@@ -230,19 +234,29 @@ func (p *envoyExtAuthzGrpcServer) eval(ctx context.Context, input ast.Value, opt
 
 		rs, err := rego.New(opts...).Eval(ctx)
 
-		// In "dry-run" mode, ignore all failure conditions
+		// In "dry-run" mode, we just log all failure conditions
 		// even ones that would typically be considered an error
-		if !p.cfg.DryRun {
-			if err != nil {
+		if err != nil {
+			if p.cfg.DryRun {
+				logrus.WithField("err", err).Warnf("Error during Rego eval.")
+			} else {
 				return err
-			} else if len(rs) == 0 {
+			}
+		} else if len(rs) == 0 {
+			if p.cfg.DryRun {
+				logrus.Warnf("Rego eval length was zero")
+			} else {
 				return fmt.Errorf("undefined decision")
-			} else if decision, ok = rs[0].Expressions[0].Value.(bool); !ok || len(rs) > 1 {
-				return fmt.Errorf("non-boolean decision")
+			}
+		} else if decision, ok = rs[0].Expressions[0].Value.(interface{}); !ok || len(rs) > 1 {
+			if p.cfg.DryRun {
+				logrus.Warnf("Decision was not interface{} OR len(rs) > 1")
+			} else {
+				return fmt.Errorf("Decision was not interface{} OR len(rs) > 1")
 			}
 		}
 
-		result.decision = decision
+		result.decision = string(util.MustMarshalJSON(decision))
```

Comment:
from the [docs](https://godoc.org/github.com/open-policy-agent/opa/util#MustMarshalJSON):

> If the data cannot be encoded, this function will panic. This function is for test purposes.

Reply:
yeah, i thought you might call me out on that... so I can just do the standard `foo, err = json.Marshal(the_bytes)` and check for error and do................... something in that case (fallback to the original value i guess?

Reply:
Fallback will set the value to empty string, so that I don't have to deal with nil checks downstream